test/test_cpp_extensions_open_device_registration.py

```diff
@@ -86,491 +99,462 @@ def setUpClass(cls):
             verbose=True,
         )
 
-    @classmethod
-    def tearDownClass(cls):
-        remove_build_path()
+        # register torch.foo module and foo device to torch
+        torch.utils.rename_privateuse1_backend("foo")
+        torch.utils.generate_methods_for_privateuse1_backend(for_storage=True)
+        torch._register_device_module("foo", generate_faked_module())
+
+    def test_base_device_registration(self):
+        self.assertFalse(self.module.custom_add_called())
+        # create a tensor using our custom device object
+        device = self.module.custom_device()
+        x = torch.empty(4, 4, device=device)
+        y = torch.empty(4, 4, device=device)
+        # Check that our device is correct.
+        self.assertTrue(x.device == device)
+        self.assertFalse(x.is_cpu)
+        self.assertFalse(self.module.custom_add_called())
+        # calls out custom add kernel, registered to the dispatcher
+        z = x + y
+        # check that it was called
+        self.assertTrue(self.module.custom_add_called())
+        z_cpu = z.to(device="cpu")
+        # Check that our cross-device copy correctly copied the data to cpu
+        self.assertTrue(z_cpu.is_cpu)
+        self.assertFalse(z.is_cpu)
+        self.assertTrue(z.device == device)
+        self.assertEqual(z, z_cpu)
```

Comment:
nit: this assertEqual is going to re-do the copy of z to the same device as z_cpu so we're only checking that the cross device copy is consistent. It's fine but I would update the comment above.

Reply:
Got it